api/v1alpha1/tlspolicy_types.go

```diff
@@ -203,6 +202,28 @@ func (p *TLSPolicy) Validate() error {
 	return nil
 }
 
+func TLSPolicyStatusMutator(desiredStatus *TLSPolicyStatus, logger logr.Logger) reconcilers.StatusMutatorFunc {
+	return func(obj client.Object) (bool, error) {
+		existing, ok := obj.(*TLSPolicy)
+		if !ok {
+			return false, fmt.Errorf("unsupported object type %T", obj)
+		}
+
+		if equality.Semantic.DeepEqual(*desiredStatus, existing.Status) {
```

Comment:
@mikenairn Here I stick to the existing implementation. What do you think about moving to `github.com/google/go-cmp/cmp` compare method with `opts` tuned for status condittions?

```go
opts := cmp.Options{                                                        
    cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),         
    cmpopts.IgnoreMapEntries(func(k string, _ any) bool {                   
        return k == "lastTransitionTime"                                    
    }),                                                                     
}                                                                           
                                                                            
if cmp.Equal(*desiredStatus, existing.Status, opts) {                    
    return false, nil                                                       
}                                                                           

```

Reply:
I don't have a strong opinion either way. If we are using `google/go-cmp/cmp` in other policies then that's probably a good enough reason to change it.

Reply:
done